rllib/utils/actors.py

```diff
@@ -65,45 +69,202 @@ def count(self):
+            (any) node.
+
+    Returns:
+        Tuple of two lists: 1) Co-located ActorHandles, 2) non co-located
+        ActorHandles.
+    """
+    if node == "localhost":
+        node = platform.node()
+
+    # Get nodes of all created actors.
     hosts = ray.get([a.get_host.remote() for a in actors])
-    local = []
-    non_local = []
-    for host, a in zip(hosts, actors):
-        if host == localhost:
-            local.append(a)
-        else:
-            non_local.append(a)
-    return local, non_local
 
+    # If `node` not provided, use the largest group of actors that sit on the
+    # same node, regardless of what that node is.
+    if node is None:
+        node_groups = defaultdict(set)
+        for host, actor in zip(hosts, actors):
+            node_groups[host].add(actor)
+        max_ = -1
+        largest_group = None
+        for host in node_groups:
+            if max_ < len(node_groups[host]):
+                max_ = len(node_groups[host])
+                largest_group = host
+        non_co_located = []
+        for host in node_groups:
+            if host != largest_group:
+                non_co_located.extend(list(node_groups[host]))
+        return list(node_groups[largest_group]), non_co_located
+    # Node provided (or default: localhost): Consider those actors "colocated"
+    # that were placed on `node`.
+    else:
+        # Split into co-located (on `node) and non-co-located (not on `node`).
+        co_located = []
+        non_co_located = []
+        for host, a in zip(hosts, actors):
+            # This actor has been placed on the correct node.
+            if host == node:
+                co_located.append(a)
+            # This actor has been placed on a different node.
+            else:
+                non_co_located.append(a)
+        return co_located, non_co_located
 
-def try_create_colocated(cls, args, count):
-    actors = [cls.remote(*args) for _ in range(count)]
-    local, rest = split_colocated(actors)
-    logger.info("Got {} colocated actors of {}".format(len(local), count))
-    for a in rest:
-        a.__ray_terminate__.remote()
-    return local
-
-
-def create_colocated(cls, args, count):
-    logger.info("Trying to create {} colocated actors".format(count))
-    ok = []
-    i = 1
-    while len(ok) < count and i < 10:
-        attempt = try_create_colocated(cls, args, count * i)
-        ok.extend(attempt)
-        i += 1
-    if len(ok) < count:
-        raise Exception("Unable to create enough colocated actors, abort.")
-    for a in ok[count:]:
+
+@Deprecated(new="create_colocated_actors", error=False)
+def create_colocated(cls, arg, count):
+    kwargs = {}
+    args = arg
+
+    return create_colocated_actors(
+        actor_specs=[(cls, args, kwargs, count)],
+        node=platform.node(),  # force on localhost
+    )[cls]
+
+
+@Deprecated(error=False)
+def drop_colocated(actors: List[ActorHandle]) -> List[ActorHandle]:
```

Comment:
delete this util? it's not used anywhere.

Reply:
Maybe some user is using it somewhere :D 
I marked it `@Deprecated`.